conkit/core/ext/_contactmap.py

```diff
@@ -40,13 +40,11 @@
 
 
 @numba.jit(nopython=True, parallel=True)
-def nb_singletons(X, threshold):
-    throwables = np.zeros(X.shape[0], dtype=np.int8)
+def nb_singletons(X, threshold, throwables):
     for i in range(X.shape[0]):
-        for j in numba.prange(i + 1, X.shape[0]):
+        for j in range(i + 1, X.shape[0]):
```

Comment:
Is `numba.prange` not an advantage here?

Reply:
`numba.prange` assumes that values in each iteration can be computed independently of the previous. Since I'm skipping all `throwables[j]` that are already flagged for removal, vectorizing this loop might cause computations where they are not needed.